spec/support/git_repo_helper.rb

```diff
@@ -0,0 +1,274 @@
+require 'rugged'
+
+module GitRepoHelper
+  # NOTE:  This ripped from the work I did in  `juliancheal/code-extractor`:
+  #
+  #     https://github.com/juliancheal/code-extractor/pull/11
+  #
+  # (from the test/test_helper.rb)
+  #
+  # Which that was a modified form of the fake_ansible_repo.rb spec helper from
+  # the ManageIQ project:
+  #
+  #     https://github.com/ManageIQ/manageiq/blob/f8e70535/spec/support/fake_ansible_repo.rb
+  #
+  # It is being repurposed here to allow generating test repos in the specs
+  # without needing to do live clones, allowing for the testing of git actions
+  # in the specs.
+  #
+  # = TmpRepo
```

Comment:
Seems a bunch of stuff is still WIP in these comments?

Reply:
I wonder if we should make this entire class it's own gem.  I think an rspec-git_repo helper gem would be awesome.

Reply:
> Seems a bunch of stuff is still WIP in these comments?

I think I mostly removed most of the parts we don't need, but I will do a once over.

> I wonder if we should make this entire class it's own gem. I think an rspec-git_repo helper gem would be awesome.

Agreed, though not sure if you want to wait on me writing another gem before we merge this :)

Reply:
I think I cleaned up anything here that might have been "extra", so I think this is good.

Aside from me creating a whole new gem for this... but ain't nobody got time for dat...